crates/sui-bytecode-src-verifier/src/lib.rs

```diff
@@ -0,0 +1,256 @@
+                // zero address is the package we're checking dependencies for
+                if verified_dependencies.contains_key(&addr) || addr.eq(&AccountAddress::ZERO) {
+                    continue;
+                }
+
+                let mod_str = module_symbol.to_string();
+                let on_chain_bytes = match on_chain_package.module_map.get(&mod_str) {
+                    Some(oc_bytes) => oc_bytes.clone(),
+                    None => {
+                        return Err(DependencyVerificationError::OnChainDependencyNotFound(
+                            pkg_symbol,
+                            Some(module_symbol),
+                        ))
+                    }
+                };
+
+                // compare local bytecode to on-chain bytecode to ensure integrity of our dependencies
+                if local_bytes != on_chain_bytes {
+                    return Err(DependencyVerificationError::ModuleBytecodeMismatch(
+                        pkg_symbol.to_string(),
+                        module_symbol.to_string(),
+                        addr,
+                    ));
+                }
+
+                on_chain_module_count += 1;
+
+                if self.verbose {
+                    println!(
+                        "{}::{} - {} bytes, code matches",
+                        pkg_symbol,
+                        module_symbol,
+                        on_chain_bytes.len()
+                    );
+                }
+            }
+
+            verified_dependencies.insert(
+                addr,
+                Dependency {
+                    symbol: pkg_symbol.to_string(),
+                    module_bytes: on_chain_package.module_map.clone(),
+                },
+            );
+        }
+
+        // total number of modules in packages must match, in addition to each individual module matching
+        let len = compiled_package.deps_compiled_units.len();
+        // only need to check for less than, because if local modules are missing on-chain we've already errored out
+        if len < on_chain_module_count {
+            let missing_modules =
+                Self::get_missing_modules(&compiled_package, &verified_dependencies);
+            return Err(DependencyVerificationError::ModuleCountMismatch(
+                len,
+                on_chain_module_count,
+                missing_modules,
+            ));
+        }
+
+        Ok(DependencyVerificationResult {
+            verified_dependencies,
+        })
+    }
+
+    fn get_missing_modules(
+        package: &CompiledPackage,
+        verified_dependencies: &HashMap<AccountAddress, Dependency>,
+    ) -> Vec<String> {
+        let mut missing_modules: Vec<String> = vec![];
+        for (local_pkg_symbol, local_unit) in &package.deps_compiled_units {
+            let local_pkg_symbol_str = local_pkg_symbol.to_string();
+            let local_mod_name = local_unit.unit.name().to_string();
```

Comment:
Could you share the error? I don't see why it would since the reference isn't leaving scope. Though you might need to bind some things to intermediate variables first.